README.md

```diff
@@ -1,16 +1,248 @@
+will invoke the method, running the code within the method. So the script above,
+saying hello five times, doing other things, then saying hello five times again
+could be rewritten as this:
+
+```ruby
+def say_hello_world_five_times
+  phrase = "Hello World!"
+  puts phrase
+  puts phrase
+  puts phrase
+  puts phrase
+  puts phrase
+end
+```
+
+That's much cleaner and follows the code principle "don't repeat yourself" or
+DRY. We abstract the action or procedure of putting "Hello World!" five times
+into a method. By defining the method `say_hello_world_five_times` once, we can
+"call" or "invoke" the method as many times as we want in the future. Let's look
+at methods in greater detail.
+
+## Define a Method
+
+Method names should begin with a lowercase letter. If you begin a method name
+with an uppercase letter, Ruby might think that it is a constant and hence can
+parse the call incorrectly.
+
+Suppose we need to define a simple method that outputs a string.
+
+You can define a method in Ruby with the `def` keyword. A method's name can
+begin with any lowercase letter. Here’s how we can do that:
+
+```ruby
+def greeting # Method Signature
+  puts "Hello World" # Method Body
+end # Method Closing
+```
+
+This defines a method. It does not use it -- yet: This is the first step so that
+it can be used later.
+
+Let’s walk through this method definition step by step:
+
+That first line, `def greeting`, is called the method signature, it defines the
+basic properties of the method including the name of the method, `greeting`.
+
+Once you open a method definition with the `def` keyword, all subsequent lines
+in your program are considered the method's body, the actual procedure or code
+that your method will run every time it's called.
+
+You must terminate every opening `def` of a method with a corresponding `end` in
+order to close the method body. If you don't correctly `end` a method, your
+program will have unexpected results or break entirely because of a syntax
+error. A good practice is to define the method and then immediately close it
+before programming anything into the method.
+
+```ruby
+def greeting
+  # Leave a line break for the method body
+end # Immediately close the method.
+```
+
+Here we set up the method's structure first, ensuring a proper termination
+before adding any other complexity. It's also a great practice to indent methods
+correctly. The body of a method should be indented two (2) spaces, placing it
+visually within the method. When you `end` the method, go back to the same
+indentation of the `def`, aligning the opening and closing of the method
+visually.
+
+Then you can easily define the body of the method and never worry about
+forgetting to `end` the method.
+
+```ruby
+def greeting
+  puts "Hello World" # Now code the body of the method.
+end
+```
+
+## Recognize How to Call Methods
+
+Once you define a method, you can execute the method whenever you want by using
```

Comment:
"Remember earlier we said, it doesn't run the work in the method _yet_. Here's where we do that..."

Tie it back to a promise you made earlier.